test/karma/providers/viewBindingsSpec.js

```diff
@@ -0,0 +1,173 @@
+describe('ViewBindings', function() {
+  beforeEach(() => window.angular.mock.module('bicker_router'));
+
+  describe('bind called with array:', () =>
+    it('get should return array', function() {
+
+      window.angular.mock.module(function(ViewBindingsProvider) {
+        ViewBindingsProvider.bind('name1', [1,2,3]);
+      });
+
+      return inject(ViewBindings => expect(ViewBindings.getView('name1').getBindings()).toEqual([1,2,3]));
+  })
+);
+
+  describe('bind called with object:', () =>
+    it('get should return one element array containing object', function() {
+
+      window.angular.mock.module(function(ViewBindingsProvider) {
+        ViewBindingsProvider.bind('name1', { 'x' : 'y' });
+      });
+
+      return inject(ViewBindings => expect(ViewBindings.getView('name1').getBindings()).toEqual([{ 'x' : 'y' }]));
+  })
+);
+
+  describe('bind called with object containing bindings array:', () =>
+    it('get should return bindings array', function() {
+
+      window.angular.mock.module(function(ViewBindingsProvider) {
+        ViewBindingsProvider.bind('name1', { 'bindings' : [1,2] });
+      });
+
+      return inject(ViewBindings => expect(ViewBindings.getView('name1').getBindings()).toEqual([1,2]));
+  })
+);
+
+  describe('bind adds commonRequiredState:', function() {
+
+    it('when commonRequiredState is a string and existing requiredState is an array', function() {
+
+      window.angular.mock.module(function(ViewBindingsProvider) {
+        ViewBindingsProvider.bind('name1', {
+          'bindings' : [{ requiredState: ['a']}, { requiredState: ['b']}],
+          'commonRequiredState': 'c'
+        }
+        );
+      });
+
+      return inject(ViewBindings => expect(ViewBindings.getView('name1').getBindings()).toEqual([{ requiredState: ['a', 'c']}, { requiredState: ['b', 'c']}]));
+  });
+
+    it('when commonRequiredState is a string and existing requiredState is a string', function() {
+
+      window.angular.mock.module(function(ViewBindingsProvider) {
+        ViewBindingsProvider.bind('name1', {
+          'bindings' : [{ requiredState: 'a'}, { requiredState: 'b'}],
+          'commonRequiredState': 'c'
+        }
+        );
+      });
+
+      return inject(ViewBindings => expect(ViewBindings.getView('name1').getBindings()).toEqual([{ requiredState: ['a', 'c']}, { requiredState: ['b', 'c']}]));
+  });
+
+    return it('when commonRequiredState is an array and existing requiredState is an array', function() {
+
+      window.angular.mock.module(function(ViewBindingsProvider) {
+        ViewBindingsProvider.bind('name1', {
+          'bindings' : [{ requiredState: ['a']}, { requiredState: ['b']}],
+          'commonRequiredState': ['c', 'd']
+        });
+      });
+
```

Comment:
I'm not too anal about single line stuff, but this is getting pretty hard to read, please tidy these up a bit - use your discretion